### DISCUSSION TITLE
Device "SleepAsAndroid" dont show up

### DISCUSSION THREAD

**Discussion by Jogi83** [IATkachenko/HA-SleepAsAndroid#41] [Q&A] [answered]
When i want to create an Automation, in the Device-Selection "SleepAs Android" dont show up.

I installed the SleepAsAndroid-Integration over HACS, and its shown up in my Integrations-Page. (i reinstalled it and i restartet my HomeAssistan serval Times)
The MQTT-Test in my App shows also "Success".

I triggered also an Event in the App, but "SleepAsAndroid" still dont show up in the Device-List from the Automation-Template .

Hope u cant help me

Thanks so far.
Greetings, Jogi

SleepAsAndroid v1.9.0
HomeAssistan v2022.4.7

**Comment by IATkachenko** [comment, marked as answer]
Jogi, hello.

What is application and the integration MQTT topic settings? 
Could you check (by MQTT Explorer, for example), that sending events from application MQTT server works well?

**Reply by Jogi83** [reply]
Hello.

First, thanks for your answer.

The Application Topic is: SleepAsAndroid
The Integration MQTT Topic is: SleepAsAndroid/%%%device%%%

I installed the MQTT-Explorer for Windows. I connected it, and when i fire up an Event in the Android-Application (for example start and then stop Sleep Tracking) it shows up that Events in the MQTT-Explorer. When i stopped it for example: SleepAsAndroid={"event":sleep_tracking_stopped"}. 

In the Blueprint i can select "Mosquitto Broker" as Device, but it should be "Sleep As Android" there, right?

Thanks and Grettings,
 Jogi

**Reply by IATkachenko** [reply]
@Jogi83, you have misconfiguration between application and the integration. The application is writing data to "SleepAsAndroid", but the integration is looking for subtopics in "SleepAsAndroid".
You have two options to solve this:
1. use topic in the **aplication** like `SleepAsAndroid/jogi83`, then device "SleepAsAndroid_jogi83" will be created (it is preferred option);
2. use topic in the **integration** `SleepAsAndroid`, then device "SleepAsAndroid_SleepAsAndroid" will be created.

> In the Blueprint i can select "Mosquitto Broker" as Device, but it should be "Sleep As Android" there, right?

Yes, you are right here.

**Reply by Jogi83** [reply]
Thank you, now I get it. I wasn't quite sure what I did wrong. The topic "HomeAssistant" is still new to me. Now it's working. Thanks very much.

**Reply by IATkachenko** [reply]
Welcome!
